pkg/iac/scanners/terraform/parser/evaluator.go

```diff
@@ -254,6 +260,9 @@ func (e *evaluator) evaluateSteps() {
 
 		e.logger.Debug("Starting iteration", log.Int("iteration", i))
 		e.evaluateStep()
+		// Always attempt to expand any blocks that might now be expandable
+		// due to new context being set.
+		e.blocks = e.expandBlocks(e.blocks)
```

Comment:
This is the biggest change. It allows `expandBlocks` to occur in `evaluateSteps`.

From my understanding of `expandBlocks` this should be a safe operation. Is there a reason it is currently called outside this context? @nikpivkin ?